### PR TITLE
CA-2653:  Nix updates & cleanups

### PR DESCRIPTION
**Pull request by deepfire**
1. Drop some of the legacy code.
2. Bump node to 1.25.1+
3. Update Nix files to match the node
4. Bump `haskell.nix` to fix an eval issue (https://github.com/input-output-hk/haskell.nix/pull/1006)

### CODE REVIEW

File: cardano-tx-generator/README.md [hunk 1]
```diff
@@ -1 +0,0 @@
-# Transaction Generator
```
Comment (MarcFontaine) [inline cardano-tx-generator/README.md]: So now we have not README.md at all ????
Why did you delete it. It did not do any harm.

File: cardano-tx-generator/cardano-tx-generator.cabal [hunk 2]
```diff
@@ -110,23 +110,3 @@ executable cardano-tx-generator
...
   default-extensions:  NoImplicitPrelude
-
-test-suite cardano-tx-generator-test
-  hs-source-dirs:       test
-  main-is:              Main.hs
-  type:                 exitcode-stdio-1.0
-
-  build-depends:        base >=4.12 && <5
-                      , tasty-hunit
-                      , tasty
```
Comment (MarcFontaine) [inline cardano-tx-generator/cardano-tx-generator.cabal]: Lets keep the test cases.
I find them useful (and I actually added new tests in my last PR).
Even if it was only for having a stub of a test suite, it should stay
If the break nix in a complicated way then lets disable them and resurrect soon.

File: cardano-tx-generator/test/Main.hs [hunk 3]
```diff
@@ -1,106 +0,0 @@
...
-                 [--addr-mainnet] 
-                 (--genesis-funds-key FILEPATH | --utxo-funds-key FILEPATH
-                   --tx-in TX-IN --tx-out TX-OUT |
-                   --split-utxo-funds-key FILEPATH --split-utxo FILEPATH)
-
-Available options:
-  --config FILEPATH        Configuration file for the cardano-node
-  --socket-path FILEPATH   Path to a cardano-node socket
-  --shelley                Initialise Cardano in shelley submode.
-  --mary                   Initialise Cardano in mary submode.
```
Comment (MarcFontaine) [inline cardano-tx-generator/test/Main.hs]: This was just added for multi era support.
Git blame on the testcase show exactly when CLI arguments change.
IHMO that is useful.

File: cardano-tx-generator/test/Main.hs [hunk 4]
```diff
@@ -1,106 +0,0 @@
...
-
-sizedMetadata = testGroup "properties of the CBOR encoding relevant for generating sized metadat"
-  [ testCase "Shelley metadata map costs" $ assertBool "metadata map costs" prop_mapCostsShelley
-  , testCase "Shelley metadata ByteString costs" $ assertBool "metadata ByteString costs" prop_bsCostsShelley
-  , testCase "Allegra metadata map costs" $ assertBool "metadata map costs" prop_mapCostsAllegra
-  , testCase "Allegra metadata ByteString costs" $ assertBool "metadata ByteString costs" prop_bsCostsAllegra
-  , testCase "Mary metadata map costs"    $ assertBool "metadata map costs" prop_mapCostsMary
-  , testCase "Marymetadata ByteString costs"    $ assertBool "metadata ByteString costs" prop_bsCostsMary
-  , testCase "Test mkMetadata" $ assertBool "" True --WIP
-  ]
```
Comment (MarcFontaine) [inline cardano-tx-generator/test/Main.hs]: This is basically what you called calibration of the sized-payload function.
Since sized payloads rely on several assumptions about the opaque particularities of the CBOR encoding.
It makes sense to at least test those assumptions.

File: default.nix [hunk 5]
```diff
@@ -21,30 +22,51 @@ let
...
+    } ''
+      cp -R ${p} $out
+      chmod -R +w $out
+      rewrite-libs $out/bin $out/bin/*
+    '' else if (pkgs.stdenv.hostPlatform.isMusl) then
+    pkgs.runCommandCC p.name { } ''
+      cp -R ${p} $out
+      chmod -R +w $out
+      $STRIP $out/bin/*
+    '' else p;
```
Comment (MarcFontaine) [inline default.nix]: I have no idea what that is?
These are the magic stub files that we just copy over from the nix-guys ?
Why do we have to deal with Darwin and Musl ?

**Review by MarcFontaine — APPROVED** [review]
OK but the test cases should be re-added (in an other PR later or in this PR).

File: cardano-tx-generator/README.md (hunk 1 above)
Comment (deepfire) [reply cardano-tx-generator/README.md]: It wasn't pulling its meaning-per-byte weight, but sure let's get it back!

File: cardano-tx-generator/cardano-tx-generator.cabal (hunk 2 above)
Comment (deepfire) [reply cardano-tx-generator/cardano-tx-generator.cabal]: Ok, agreed.

File: cardano-tx-generator/test/Main.hs (hunk 4 above)
Comment (deepfire) [reply cardano-tx-generator/test/Main.hs]: Oh, wow, my apologies @MarcFontaine, I was really, _really_ too quick over here!

Thank you for catching this!

File: default.nix (hunk 5 above)
Comment (deepfire) [reply default.nix]: I've cut off a _lot_ of darwin/musl stuff from the `cardano-node` blueprint, but that still slipped through.

Thank you Marc!

File: cardano-tx-generator/test/Main.hs (hunk 3 above)
Comment (deepfire) [reply cardano-tx-generator/test/Main.hs]: I'm of a split mind here -- this really increases the cost of change.

And in a recent discussion we've agreed that it's _already_ too painful to change the CLI, if you remember.
